app/src/main/java/edu/wisc/ece/pinpoint/MainActivity.java

```diff
@@ -3,57 +3,52 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.MenuItem;
+import android.view.View;
 
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+import androidx.navigation.ui.NavigationUI;
 
 import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
 import com.google.android.material.navigation.NavigationBarView;
 
 import edu.wisc.ece.pinpoint.utils.FirebaseDriver;
 
 public class MainActivity extends AppCompatActivity {
     private FirebaseDriver firebase;
-    private BottomNavigationView navBar;
+    private FloatingActionButton mapButton;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         firebase = FirebaseDriver.getInstance();
-        navBar = findViewById(R.id.navBar);
+        NavController navController = Navigation.findNavController(this, R.id.activity_main_nav_host_fragment);
+        BottomNavigationView navBar = findViewById(R.id.navBar);
         navBar.getMenu().getItem(2).setEnabled(false);
-        navBar.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
+        NavigationUI.setupWithNavController(navBar, navController);
+        mapButton = findViewById(R.id.mapButton);
+        mapButton.setOnClickListener(new View.OnClickListener() {
             @Override
-            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
-                switch(item.getItemId()){
-                    case R.id.navbarProfile:
-                        startActivity(new Intent(getApplicationContext(), ProfilePageActivity.class));
-                        overridePendingTransition(0,0);
-                        return true;
-                    case R.id.navbarSearch:
-                        return true;
-                    case R.id.navbarLeaderboard:
-                        return true;
-                    case R.id.navbarFeed:
-                        return true;
-                }
-                return false;
+            public void onClick(View view) {
+                navController.navigate(R.id.navbarMap);
             }
         });
     }
 
     @Override
     public void onStart() {
         super.onStart();
-        navBar.setSelectedItemId(R.id.navbarEmpty);
         // Check if user is not signed in, if so start auth flow
         if (firebase.getUser() == null) {
-            firebase.launchAuth(this);
+            //firebase.launchAuth(this);
         } else {
             // User is logged in!
             // Log user out for testing purposes
-            firebase.logout(this);
+            //firebase.logout(this);
```

Comment:
Can we uncomment this or is it supposed to stay commented?

Reply:
This was for debugging purposes. I'll uncomment them